### PR TITLE
Mostly working first version of DOI info update/entry handling code

### LINKED ISSUE #157: Add code to store and update DOI information
Adding in functionality to handle DOI registration information as it is sent from the Minting API (along with updates)

### PR DESCRIPTION
**Pull request by kauberry**
### Description

Adding a REST endpoint to receive commands for updating DOI entry data

### Issues Resolved

#157 

### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: metadata/rest/doi_queries/doi_registration_update.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        }
+        doi_trans_mapping, mapping_created = DOITransaction.get_or_create(**doi_trans_map_item)
+        return doi_trans_mapping.doi.doi
+
+    @staticmethod
+    def _get_updated_osti_info(doi_string, elink_url, elink_user, elink_password):
+        """This will access the server at OSTI and get all the relevant details for this DOI."""
+        osti_elink_url = elink_url + '?doi=' + doi_string
+
+        r = requests.get(osti_elink_url, auth=(elink_user, elink_password))
```
Comment (dmlb2000) [inline metadata/rest/doi_queries/doi_registration_update.py]: I'm not sure I like this option, doesn't the DOI service already have this info from osti? Can't the DOI service send me this info from osti in the post?

File: pacifica/metadata/rest/doi_queries/doi_registration_update.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    # CherryPy requires these named methods.
+    # pylint: disable=invalid-name
+    @staticmethod
+    @tools.json_out()
+    def POST():
+        """Update existing DOI Entries."""
+        valid_content_types = ['application/xml', 'text/xml']
+        headers = request.headers
+        if headers['Content-type'] not in valid_content_types:
+            raise HTTPError(415, 'Expected an entity of content type %s' %
```
Comment (dmlb2000) [inline pacifica/metadata/rest/doi_queries/doi_registration_update.py]: Just a comment for future note. Seems CherryPy has some built in XML support but that effort is around XML-RPC specifically and they don't provide clean tools similar to their JSON in and out methods.

File: tests/test_files/osti_update.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?xml version='1.0' encoding='utf-8'?>
```
[the remark this answers is not in the document]
Comment (kauberry) [reply tests/test_files/osti_update.xml]: I can't manage to find any XSD's for the return format. I'll contact the technical guys at OSTI and ask if they can provide one.

[the remark this answers is not in the document]
Comment (kauberry) [reply tests/test_files/osti_update.xml]: Barring that, I'll have a look at wrapping the whole thing in an atomic transaction anyway.

File: pacifica/metadata/rest/doi_queries/doi_registration_update.py (hunk 2 above)
Comment (kauberry) [reply pacifica/metadata/rest/doi_queries/doi_registration_update.py]: I looked for a built-in xml handler like the one for json. That would have been nice to have. The RPC stuff was pretty unwieldy.

File: requirements.txt [hunk 4]
```diff
@@ -5,3 +5,4 @@ psycopg2
 python-dateutil
 requests
 six
+urllib3==1.23
```
[the remark this answers is not in the document]
Comment (kauberry) [reply requirements.txt]: I thought it was rather strict as well, but it seems that 1.24 breaks requests maybe? Perhaps docker was just bitchy about it.

[the remark this answers is not in the document]
Comment (kauberry) [reply requirements.txt]: I've pulled out the urllib requirement for now. Let's see how things react.

**Comment by kauberry** [comment]
This should get all the changes in save the XSD, but I've got the db.atomic bits in place for create/update.

**Pull request merged**